### PR TITLE
fix: setting builder optimization level to TRT 8.6 default

### PR DESCRIPTION
**Pull request by gedoensmax**
The actual released default level is 3 and not the previously used 2.

Just a small sample of the effects:
Screenshot 2023-05-10 at 15 49 55

### CODE REVIEW

**Comment by chilo-ms** [comment]
Could you help merge the main to pass "Linux GPU TensorRT CI Pipeline"? 
There are some PRs related to ONNX 1.14 release were merged right after your PR.

File: onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc
Comment (chilo-ms) [inline onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc:1]: line 2831 & 2892 need to be modified as well.

`if (trt_state->builder_optimization_level != 2) {`

**Review by jywu-msft — APPROVED** [review]

File: onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc
Comment (jywu-msft) [reply onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc:1]: we'll have to kick off the CI again after those lines are updated.

Comment (gedoensmax) [reply onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc:1]: Done! Is this something your team is working on right now ?

Comment (jywu-msft) [reply onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc:1]: it's on our radar and we would like to dedicate some time/resources on it.

**Pull request merged**